File: HelloWorld/procedural_3.rb
#!/usr/bin/env ruby
# encoding: utf-8

MESSAGE = 'Hello, %s!'

def greet( message, name = 'World' )
  message % name
end

# Array.size() is not procedural: it's a method of the object Array.
greeting = ARGV.size != 0 ? greet( MESSAGE, name = ARGV[ 0 ] ) : greet( MESSAGE )
puts greeting

puts '» José Carlos Monteiro «'

File: HelloWorld/procedural_3.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# https://www.python.org/dev/peps/pep-0263/

# Must import module "sys" to handle command line arguments ( which is not procedural ).
import sys

MESSAGE = 'Hello, %(name)s!'

def greet( message, name = 'World' ) :
  message % { 'name' : name }

if __name__ == '__main__' :

  greeting = greet( MESSAGE, sys.argv[ 1 ] ) if len( sys.argv ) - 1 != 0 else greet( MESSAGE )
  print( greeting )

  print( '» José Carlos Monteiro «' )
